### DISCUSSION TITLE
Fabric in react and saving into database

### DISCUSSION THREAD

**Discussion by Paolo-Federle** [fabricjs/fabric.js#8625] [Q&A]
Hello everyone.
I'm am very new to the industry and i am trying to use this library for my first big project but I'm a bit struggling with the documentations and tutorial to get everything work as intended.

https://codesandbox.io/s/zen-dirac-kctx8s?file=/src/App.js

This is a component from my project.
At the moment I was able to create a canvas, simple forms and even images from the web.
But I wasn't able to add the functionality to delete something by pressing delete or canc.
I was even trying to save the canvas to a database, each time something get added, so that if someone get to that specific canvas later can still see what was done but each time I try to initiate the canvas with the saved content it doesn't work.
I suspect that may be for some basic setting of my code, even because I needed a button to initiate the canvas and I'm failing to create a canvas using useEffect or any other stuff when the page load for the first time.

Probably I'm asking for really basic stuff but as a beginner I'm trying to understand how to use this library.
thank you

**Comment by iocoker** [comment]
Generally, you need to handle keyboard events and determine the action to apply based on the [keyCodes](https://www.toptal.com/developers/keycode). So, for Delete `46`, you'll get the Canvas' active object and remove it like so: `fabricCanvas.remove(fabricCanvas.getActiveObject())` of course, you will check if there's an activeObject selected in the first place. And don't forget to render your canvas after the deletion.

Same rule for depth management, moving objects with arrow keys etc. Just get your `activeObject` and apply whatever action on it and render the canvas.

For saving to database, it's pretty straight forward too,  `fabricCanvas.toObject()` will generate the complete JSON object you need to send to your database. And to load it back, you use [loadFromJSON](http://fabricjs.com/docs/fabric.Canvas.html#loadFromJSON)
So, first you create the canvas, then call loadFromJSON on it, passing the JSON string.

Another thing you should be wary of is, React Proxying your objects. In VueJS, I normally use markRaw to prevent VueJS from taking hold of and tracking a fabric object: https://github.com/fabricjs/fabric.js/issues/7012#issuecomment-1117022231

Unfortunately I don't do React, but the same general ideas applies to most frameworks.
